examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,297 @@
+    image = preprocess_input(image)
+    return image
+
+
+def data_preprocess(source, target):
+    source_image = preprocess(source)
+    target_image = preprocess(target)
+    return source_image, target_image
+
+
+def denormalize(array):
+    array += 1
+    array *= 127.5
+    return array
+
+
+buffer_size = int(len(train_source) * 0.2)
+train_pipe = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
+train_pipe = train_pipe.map(data_preprocess, tf.data.AUTOTUNE).shuffle(buffer_size)
+train_pipe = train_pipe.batch(batch_size).repeat()
+
+val_pipe = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+val_pipe = val_pipe.map(data_preprocess, tf.data.AUTOTUNE).shuffle(buffer_size)
+val_pipe = val_pipe.batch(batch_size).repeat()
+
+
+"""
+## Dataset Visualiztion
+"""
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(val_pipe.take(1)):
+    noise, clean = f
+    noise = denormalize(noise[i].numpy())
+    clean = denormalize(clean[i].numpy())
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+
+We have used Pre-trained ```ResNet50V2``` U-Net as Encoder, The Feature Maps are
+extracted at different levels forming Downscaling path, Then ```Sub-Pixel Layers``` on
+the Upscaling path with skip-connections.
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
+    base_model = ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(352, 608, 3)
+    )
+    encode0 = base_model.get_layer("input_1").output
+    encode1 = base_model.get_layer("conv1_conv").output
+    encode2 = base_model.get_layer("conv2_block3_1_relu").output
+    encode3 = base_model.get_layer("conv3_block4_1_relu").output
```

Comment:
@anish9 It would be nice to show the output size of each intermediate layer in a comment, for example 

```python
encode1 = base_model.get_layer("conv1_conv").output # (None, h, w, depth) 
...
```